workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/SpatialOSBehaviourManager.cs

```diff
@@ -1,60 +1,227 @@
+            spatialId = spatialComponent.SpatialEntityId;
+
+            foreach (var behaviour in gameObject.GetComponents<MonoBehaviour>())
+            {
+                var readerIds = library.GetRequiredReaderComponentIds(behaviour.GetType());
+                foreach (var id in readerIds)
+                {
+                    if (!behavioursRequiringReaderTypes.TryGetValue(id, out var behavioursForReader))
+                    {
+                        behavioursForReader = new HashSet<MonoBehaviour>();
+                        behavioursRequiringReaderTypes[id] = behavioursForReader;
+                    }
+
+                    behavioursForReader.Add(behaviour);
+                }
+
+                var writerIds = library.GetRequiredWriterComponentIds(behaviour.GetType());
+                foreach (var id in writerIds)
+                {
+                    if (!behavioursRequiringWriterTypes.TryGetValue(id, out var behavioursForWriter))
+                    {
+                        behavioursForWriter = new HashSet<MonoBehaviour>();
+                        behavioursRequiringWriterTypes[id] = behavioursForWriter;
+                    }
+
+                    behavioursForWriter.Add(behaviour);
+                }
+
+                numUnsatisfiedReadersOrWriters[behaviour] = readerIds.Count + writerIds.Count;
+            }
         }
 
-        public List<ReaderWriterImpl> GetReadersWriters(uint componentId)
+
+        public HashSet<IReaderInternal> GetReadersWriters(uint componentId)
         {
-            return null;
+            return compIdToReadersWriters[componentId];
         }
 
         public void EnableSpatialOSBehaviours()
         {
-            // inject + enable all spatialOSBehavioursToEnable
+            foreach (var behaviour in behavioursToEnable)
+            {
+                var dict = behaviourLibrary.InjectAllReadersWriters(behaviour, entity);
+                behaviourToReadersWriters[behaviour] = dict;
+                foreach (var idToReaderWriter in dict)
+                {
+                    var id = idToReaderWriter.Key;
+                    var reader = idToReaderWriter.Value;
+                    if (!compIdToReadersWriters.ContainsKey(id))
+                    {
+                        compIdToReadersWriters[id] = new HashSet<IReaderInternal>();
+                    }
+
+                    compIdToReadersWriters[id].Add(reader);
+                }
+
+                try
+                {
+                    behaviour.enabled = true;
+                }
+                catch (Exception e)
+                {
+                    logger.HandleLog(LogType.Error,
+                        new LogEvent("Exception thrown in OnEnable() method of MonoBehaviour.")
+                            .WithField(LoggingUtils.LoggerName, LoggerName)
+                            .WithField(LoggingUtils.EntityId, spatialId)
+                            .WithField("Exception", e.ToString()));
```

Comment:
Can you pass the full exception object to the logger, rather than the `ToString`?